src/plugins/intel_gpu/src/graph/col2im.cpp

```diff
@@ -45,6 +38,24 @@ std::vector<layout> col2im_inst::calc_output_layouts(col2im_node const& node, ke
     auto output_type = desc->output_data_types[0].value_or(input_layout.data_type);
     auto output_format = input_layout.format;
 
+    auto reshaped_input = input_layout;
+    if (input_layout.get_rank() >= 4) {
+        bool is_batched = true;
+        auto num_blocks_l = input_layout.spatial(1);
```

Comment:
I'm afraid such statement would fail for dynamic shape. Could you add dynamic shape test as well?